drivers/platform/xilinx/spi.c

```diff
@@ -179,9 +183,21 @@ static int32_t spi_init_ps(struct spi_desc *desc,
 	if(ret != SUCCESS)
 		goto ps_error;
 
+	switch (xinit->device_id) {
+	case 0:
+		input_clock = SPI_CLK_FREQ_HZ(0);
+		break;
+	case 1:
+		input_clock = SPI_CLK_FREQ_HZ(1);
+		break;
```

Comment:
I would do:
case 0:
case 1:
   input_clock = SPI_CLK_FREQ_HZ(xinit->device_id);
   break;
However, is there any reason for limiting this to 0 and 1?

Reply:
It won't work like that, because of what SPI_CLK_FREQ_HZ does. It uses the preprocessor directive ## to literally combine strings.

I checked the PS7 and PSU architecture manual and both are limited only to SPI0 and SPI1 so this switch should be ok. In case we wlil support a new architecture that will have SPI2, SPI3 etc. we will need to extend the switch.

Reply:
OK. Thanks.